pipenv/core.py

```diff
@@ -713,6 +712,7 @@ def batch_install(deps_list, procs, failed_deps_queue,
     trusted_hosts = []
     # Install these because
     for dep in deps_list_bar:
+        install_deps = not no_deps
```

Comment:
so this decision was effectively being overridden? I'm trying to reason about how that would cause the issue in question but I'm not really sure I understand yet

Reply:
IIRC, the problem occurred when there was at least one dependency fulfilling this condition `not PIPENV_RESOLVE_VCS and is_artifact and not dep.editable` (installed with `git://`?) and all the consecutive deps were failing due to this

Reply:
@techalchemy and as far as I can see the current `master` is still affected by this issue

Reply:
`is_artifact` is true for anything that is like a `zip` or `whl` or a source control path that is not in editable mode.

Reply:
ah sorry I didn't have the code expanded for my first comment, I can see the issue here -- `install_deps` is what we actually pass to `pip_install`, but it gets re-defined under specific conditions (i.e. if `not PIPENV_RESOLVE_VCS and is_artifact and not dep.editable`).

Because it is sometimes altered during a pass of the loop, you chose to reset it at the start of each iteration, which makes sense., and git rid of the now-unused `no_deps` variable. This change makes a lot of sense to me.